stdlib/public/core/RangeSet.swift

```diff
@@ -0,0 +1,376 @@
+  }
+}
+
+// MARK: - SetAlgebra
+
+// These methods only depend on the ranges that comprise the range set, so
+// we can provide them even when we can't provide `SetAlgebra` conformance.
+@available(SwiftStdlib 5.11, *)
+extension RangeSet {
+  /// Adds the contents of the given range set to this range set.
+  ///
+  /// - Parameter other: A range set to merge with this one.
+  public mutating func formUnion(_ other: __owned RangeSet<Bound>) {
+    for range in other._ranges {
+      insert(contentsOf: range)
+    }
+  }
+  
+  /// Removes the contents of this range set that aren't also in the given
+  /// range set.
+  ///
+  /// - Parameter other: A range set to intersect with.
+  public mutating func formIntersection(_ other: RangeSet<Bound>) {
+    self = self.intersection(other)
+  }
+  
+  /// Removes the contents of this range set that are also in the given set
+  /// and adds the contents of the given set that are not already in this
+  /// range set.
+  ///
+  /// - Parameter other: A range set to perform a symmetric difference against.
+  public mutating func formSymmetricDifference(
+    _ other: __owned RangeSet<Bound>
+  ) {
+    self = self.symmetricDifference(other)
+  }
+  
+  /// Removes the contents of the given range set from this range set.
+  ///
+  /// - Parameter other: A range set to subtract from this one.
+  public mutating func subtract(_ other: RangeSet<Bound>) {
+    for range in other._ranges {
+      remove(contentsOf: range)
+    }
+  }
+  
+  /// Returns a new range set containing the contents of both this set and the
+  /// given set.
+  ///
+  /// - Parameter other: The range set to merge with this one.
+  /// - Returns: A new range set.
+  public __consuming func union(
+    _ other: __owned RangeSet<Bound>
+  ) -> RangeSet<Bound> {
+    var result = self
+    result.formUnion(other)
+    return result
+  }
+  
+  /// Returns a new range set containing the contents of both this set and the
+  /// given set.
+  ///
+  /// - Parameter other: The range set to merge with this one.
+  /// - Returns: A new range set.
+  public __consuming func intersection(
+    _ other: RangeSet<Bound>
+  ) -> RangeSet<Bound> {
+    return RangeSet(_ranges: _ranges._intersection(other._ranges))
+  }
+  
+  /// Returns a new range set representing the values in this range set or the
+  /// given range set, but not both.
+  ///
+  /// - Parameter other: The range set to find a symmetric difference with.
+  /// - Returns: A new range set.
+  public __consuming func symmetricDifference(
+    _ other: __owned RangeSet<Bound>
+  ) -> RangeSet<Bound> {
+    return union(other).subtracting(intersection(other))
+  }
+  
+  /// Returns a new set containing the contents of this range set that are not
+  /// also in the given range set.
+  ///
+  /// - Parameter other: The range set to subtract.
+  /// - Returns: A new range set.
+  public func subtracting(_ other: RangeSet<Bound>) -> RangeSet<Bound> {
```

Comment:
I think we lost a `__consuming` qualifier here.

```suggestion
  public __consuming func subtracting(
    _ other: RangeSet<Bound>
  ) -> RangeSet<Bound> {
```